sim/battle.ts

```diff
@@ -1055,12 +1055,32 @@ export class Battle {
 		return handlers;
 	}
 
+	getCallback(target: Pokemon | Side | Field | Battle, effect: Effect, callbackName: string, getKey?: 'duration') {
+		// @ts-expect-error dynamic lookup
+		let callback = effect[callbackName];
```

Comment:
```suggestion
		let callback: Function | undefined = (effect as any)[callbackName];
```
These days I would use this rather than expect-error. Probably we can find a better type than `Function`, too.